src/app/services/auth/auth.service.ts

```diff
@@ -1,26 +1,166 @@
+  @LocalStorage() token2: string;
+  private loggedInUser$ = new BehaviorSubject<LoggedInUser>(null);
+
+  private isLoggedIn$ = new BehaviorSubject<boolean>(false);
+  get isAuthenticated(): boolean {
+    return this.isLoggedIn;
+  }
+
+  get isAuthenticated$(): Observable<boolean> {
+    return this.isLoggedIn$.asObservable();
+  }
+
+  get user$(): Observable<LoggedInUser> {
+    return this.loggedInUser$.asObservable();
+  }
 
-@Injectable()
-export class AuthService implements CanActivate {
   constructor(
-    private router: Router,
-    private ws: WebSocketService,
-    @Inject(WINDOW) private window: Window,
+    private wsManager: WebsocketManagerService,
   ) { }
 
-  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean {
-    if (this.ws.loggedIn) {
-      return true;
-    }
+  login(username: string, password: string): Observable<boolean> {
+    const uuid = UUID.UUID();
+    this.wsManager.send({
+      id: uuid,
+      msg: IncomingApiMessageType.Method,
+      method: 'auth.login',
+      params: [username, password],
+    });
+    return this.getFilteredWebsocketResponse<boolean>(uuid).pipe(tap((response) => {
+      this.isLoggedIn = response;
+      this.isLoggedIn$.next(response);
+    }));
+  }
+
+  loginWithOtp(username: string, password: string, otp: string): Observable<boolean> {
+    const uuid = UUID.UUID();
+    this.wsManager.send({
+      id: uuid,
+      msg: IncomingApiMessageType.Method,
+      method: 'auth.login',
+      params: [username, password, otp],
+    });
+    return this.getFilteredWebsocketResponse<boolean>(uuid).pipe(tap((response) => {
+      this.isLoggedIn = response;
+      this.isLoggedIn$.next(response);
+    }));
+  }
+
+  loginWithToken(): Observable<boolean> {
+    const uuid = UUID.UUID();
+    this.wsManager.send({
+      id: uuid,
+      msg: IncomingApiMessageType.Method,
+      method: 'auth.login_with_token',
+      params: [this.token2],
+    });
+    return this.getFilteredWebsocketResponse<boolean>(uuid).pipe(tap((response) => {
+      this.isLoggedIn = response;
+      this.isLoggedIn$.next(response);
+    }));
+  }
+
+  getFilteredWebsocketResponse<T>(uuid: string): Observable<T> {
+    return this.wsManager.websocketSubject$.pipe(
+      filter((data: IncomingWebsocketMessage) => data.msg === IncomingApiMessageType.Result && data.id === uuid),
+      map((data: ResultMessage<T>) => data.result),
+      take(1),
+    );
+  }
+
+  generateToken(tokenLiftime: number): Observable<string> {
+    const uuid = UUID.UUID();
+    const payload: {
+      id: string;
+      msg: IncomingApiMessageType;
+      method: string;
+      params?: [number];
+    } = {
+      id: uuid,
+      msg: IncomingApiMessageType.Method,
+      method: 'auth.generate_token',
+      params: [tokenLiftime],
+    };
+    this.wsManager.send(payload);
+    return this.getFilteredWebsocketResponse<string>(uuid);
+  }
+
+  generateTokenWithDefaultLifetime(): Observable<string> {
```

Comment:
This method is not called anywhere.

Reply:
It is on line 112 of `TerminalComponent`